lib/puppetserver/ca/sign_action.rb

```diff
@@ -0,0 +1,252 @@
+            exit_code = 1
+          else
+            @logger.err 'Error - gross debug output:'
+            @logger.err "    #{result.inspect}"
+            exit_code = 1
+          end
+        end
+        return exit_code
+      end
+
+      def store
+        unless @store
+          store = OpenSSL::X509::Store.new
+          store.add_file(File.expand_path('~/.puppetlabs/etc/puppet/ssl/certs/ca.pem'))
+          store.add_file(File.expand_path('~/.puppetlabs/etc/puppet/ssl/certs/localhost.pem'))
+          store.add_file(File.expand_path('~/.puppetlabs/etc/puppet/ssl/crl.pem'))
+          @store = store
+        end
+        return @store
+      end
+
+      def check_flag_usage(results,parser)
+        if results['certname'] && results['all']
+          @logger.err '--all and --certname cannot be used together'
+          @logger.err parser.help
+          return false
+        elsif !results['certname'] && !results['all']
+          @logger.err 'No arguments given'
+          @logger.err parser.help
+          return false
+        elsif results['certname'] && results['certname'].include?('--all')
+          @logger.err 'Cannot use --all with --certname. If you actually have a certificate request ' +
+                          'for a certifcate named --all, you need to use the HTTP API.'
+          return false
+        end
+        return true
+      end
+
+      def log_unknown_arg(unparsed,parser)
+        @logger.err 'Error:'
+        @logger.err 'Unknown arguments or flags:'
+        unparsed.each do |arg|
+          @logger.err "    #{arg}"
+        end
+
+        @logger.err ''
+        @logger.err parser.help
+      end
+
+      def log_invalid_certname()
+        @logger.err 'Error:'
+        @logger.err 'A certname to sign is required'
+      end
+
+      def check_unparsed(unparsed,parser)
+        unless unparsed.empty?
+          log_unknown_arg(unparsed,parser)
+          return false
+        end
+        return true
+      end
+
+      def valid?(invalid,unparsed,results,parser)
+        unless invalid.empty?
+          log_invalid_certname
+          return false
+        else
+           [check_flag_usage(results,parser), check_unparsed(unparsed,parser)].all?
+        end
+      end
+
+      def parse(args)
+        results = {}
+        unparsed = []
+        invalid = []
+        parser = self.class.parser(results)
+        begin
+          parser.order!(args) do |unknown|
+            unparsed << unknown
+          end
+        rescue OptionParser::InvalidArgument => e
+          invalid += e.args
+          retry
+        rescue OptionParser::ParseError => e
+          unparsed += e.args
+          unparsed << args.shift unless args.first =~ /^-{1,2}/
+          retry
+        end
+
+        if valid?(invalid,unparsed,results,parser)
+          exit_code = nil
+        else
+          exit_code = 1
+        end
+        return results, exit_code
+      end
+
+      def parse_inputs(inputs)
```

Comment:
what's this guy doing?

Reply:
nothing, apparently  😂I think it's left over from refactoring/moving things around. Good catch!